src/extension.ts

```diff
@@ -903,6 +904,26 @@ context.subscriptions.push(disposable);
     true
   ); // Required to enable run query button at the start
 
+  context.subscriptions.push(
+    vscode.commands.registerCommand(Commands.explainQuery, async () => {
+      vscode.commands.executeCommand(
+        "setContext",
+        "vscode-couchbase.runButtonEnabled",
+        undefined
+      );
+      await workbench.runCouchbaseQuery(
+        workbenchWebviewProvider,
+        queryHistoryTreeProvider,
+        true
+      );
+      vscode.commands.executeCommand(
+        "setContext",
+        "vscode-couchbase.runButtonEnabled",
+        true
+      );
+    })
+  );
```

Comment:
Make sure, explain button is also disabled if query is running

Reply:
I haven't tested this though

Reply:
I have tested it, when we hover when running a query, explain query button cannot be clicked & vice versa